components/GraphDiagram/graph_util.ts

```diff
@@ -294,11 +296,12 @@ export function foreachDownstream(
   callback: (module: Module) => void,
   seen: Set<Module> = new Set(),
 ) {
-  const info = graph.moduleInfos.get(module.key);
-  if (!info || seen.has(module)) return;
   seen.add(module);
+  const info = graph.moduleInfos.get(module.key);
+  if (!info) return;
 
   for (const { module } of info.downstream) {
+    if (seen.has(module)) continue;
```

Comment:
The cache is handled differently here so that we only call the `callback` if the module was not already seen. Before it was calling the `callback` despite already seen.